test_runner/fixtures/pageserver/utils.py

```diff
@@ -350,6 +351,30 @@ def list_prefix(
     return response
 
 
+def remote_storage_delete_key(
+    remote: RemoteStorage,
+    key: str,
+) -> DeleteObjectOutputTypeDef:
+    """
+    Note that this function takes into account prefix_in_bucket.
+    """
+    # For local_fs we need to use a different implementation. As we don't need local_fs, just don't support it for now.
+    assert isinstance(remote, S3Storage), "localfs is currently not supported"
+    assert remote.client is not None
```

Comment:
Is the client really optional? I don't remember doing that.

Reply:
No it's not Optional, however asserting that something is not None is always ok pythonwise, but extra.

```suggestion
```

did pyright have an issue with this type? If so, interesting.

Reply:
The client is not present in localfs. This is just another safety check.

Reply:
I've removed it from all relevant places now